### PR TITLE
[ISSUE-399] pod restart node id missing

### PR DESCRIPTION
**Pull request by mishoyama**
### Issue #399

- Obtain node ID with retries and exit when threshold is reached
- `waitForVolumeManagerReadiness` function refactored - extra variables removed

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
Tested manually - no pods restart observed. Custom CI passed.

### CODE REVIEW

File: cmd/node/main.go [hunk 1]
```diff
@@ -197,12 +199,30 @@ func main() {
 	logger.Info("Got SIGTERM signal")
 }
 
+func obtainNodeIDWithRetries(client k8sClient.Client, featureConf featureconfig.FeatureChecker,
+	logger *logrus.Logger) (string, error) {
+	var (
```
Comment (rkiselenko) [inline cmd/node/main.go:204]: Just use variables in `(nodeID string, err error)`, no need variable declaration

**Review by rkiselenko — APPROVED** [review]

File: cmd/node/main.go (hunk 1 above)
Comment (mishoyama) [reply cmd/node/main.go:204]: Do you mean in `if` statement?

Comment (rkiselenko) [reply cmd/node/main.go:204]: No, I mean you can use [named resutlt parameters](https://golang.org/doc/effective_go#named-results):
```golang
func obtainNodeIDWithRetries(client k8sClient.Client, featureConf featureconfig.FeatureChecker,
   logger *logrus.Logger) (nodeID string, err error) {
	for i := 0; i < numberOfRetries; i++ {
        // other code goes here
```

Comment (mishoyama) [reply cmd/node/main.go:204]: Ok, thanks! Refactored.

**Review by safronovD — APPROVED** [review]

**Pull request merged**